src/ensembl/src/shared/state/ens-object/ensObjectActions.ts

```diff
@@ -17,107 +17,152 @@
-) => {
-  if (typeof payload === 'string') {
-    payload = parseEnsObjectId(payload);
-  }
-  const state = getState();
-  const ensObjectId = buildEnsObjectId(payload);
-  const ensObjectLoadingStatus = getEnsObjectLoadingStatus(state, ensObjectId);
-  if (!shouldFetch(ensObjectLoadingStatus)) {
-    return;
+const geneQuery = gql`
+  query Gene($genomeId: String!, $geneId: String!) {
+    gene(byId: { genome_id: $genomeId, stable_id: $geneId }) {
+      name
+      stable_id
+      unversioned_stable_id
+      symbol
+      slice {
+        region {
+          name
+        }
+        location {
+          start
+          end
+        }
+        strand {
+          code
+        }
+      }
+      metadata {
+        biotype {
+          label
+        }
+      }
+      transcripts {
+        stable_id
+        slice {
+          location {
+            length
+          }
+        }
+        product_generating_contexts {
+          product_type
+        }
+        metadata {
+          biotype {
+            label
+          }
+          canonical {
+            value
+            label
+          }
+          mane {
+            value
+            label
+            ncbi_transcript {
+              id
+              url
+            }
+          }
+          gencode_basic {
+            label
+          }
+        }
+      }
+    }
   }
+`;
 
-  if (payload.type === 'region') {
-    const regionObject = buildRegionObject(payload);
-    dispatch(
-      fetchEnsObjectAsyncActions.success({
-        id: ensObjectId,
-        data: regionObject
-      })
+export const fetchEnsObject =
+  (
+    payload: string | EnsObjectIdConstituents
+  ): ThunkAction<void, any, null, Action<string>> =>
+  async (dispatch, getState: () => RootState) => {
+    if (typeof payload === 'string') {
+      payload = parseEnsObjectId(payload);
+    }
+    const state = getState();
+    const ensObjectId = buildEnsObjectId(payload);
+    const ensObjectLoadingStatus = getEnsObjectLoadingStatus(
+      state,
+      ensObjectId
     );
-    return;
-  }
-
-  try {
-    dispatch(fetchEnsObjectAsyncActions.request(ensObjectId));
-    const { genomeId, objectId, type } = payload;
-
-    const objectInfoUrl = `/api/genomesearch/object/info?genome_id=${genomeId}&type=${type}&stable_id=${objectId}`;
-    const objectTracksUrl = `/api/genomesearch/object/track_list?genome_id=${genomeId}&type=${type}&stable_id=${objectId}`;
-    const response: EnsObjectResponse = await apiService.fetch(objectInfoUrl);
-    response.object_id = buildEnsObjectId(payload);
+    if (!shouldFetch(ensObjectLoadingStatus)) {
+      return;
+    }
 
-    try {
-      response.track = await apiService.fetch(objectTracksUrl);
-    } catch {
-      // FIXME: this is a temporary solution
-      response.track = builtTrackList(response);
+    if (payload.type === 'region') {
+      const regionObject = buildRegionObject(payload);
+      dispatch(
+        fetchEnsObjectAsyncActions.success({
```

Comment:
I think this way of fetching the data is okay but I am not sure if we still need to use `request`, `success` & `failure` actions.

Reply:
But isn't that the standard we are using at the moment for async actions?